### PR TITLE
Updates k8s stack

### PR DESCRIPTION
**Pull request by f41gh7**

### CODE REVIEW

File: charts/victoria-metrics-k8s-stack/templates/servicemonitors/kubedns.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+{{- if .Values.kubeDns.service.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ printf "%s-%s" (include "victoria-metrics-k8s-stack.fullname" $) "kubedns" | trunc 63 | trimSuffix "-" }}
+  namespace: kube-system
+  labels:
+    app: {{ include "victoria-metrics-k8s-stack.fullname" . }}-kubedns
+    jobLabel: kubedns
+  {{ include "victoria-metrics-k8s-stack.labels" . | indent 4 }}
```
Comment (tenmozes) [inline charts/victoria-metrics-k8s-stack/templates/servicemonitors/kubedns.yaml:11]: this looks weird 2 spaces that indent 4, could we use only one approach

File: charts/victoria-metrics-k8s-stack/templates/servicemonitors/kubedns.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - name: http-metrics-dnsmasq
+      port: {{ .Values.kubeDns.service.dnsmasq.port }}
+      protocol: TCP
+      targetPort: {{ .Values.kubeDns.service.dnsmasq.targetPort }}
+    - name: http-metrics-skydns
+      port: {{ .Values.kubeDns.service.skydns.port }}
+      protocol: TCP
+      targetPort: {{ .Values.kubeDns.service.skydns.targetPort }}
+  selector:
+    {{- if .Values.kubeDns.service.selector }}
```
Comment (tenmozes) [inline charts/victoria-metrics-k8s-stack/templates/servicemonitors/kubedns.yaml:24]: please change to with

File: charts/victoria-metrics-k8s-stack/values.yaml [hunk 3]
```diff
@@ -737,6 +737,31 @@ kubeControllerManager:
             caFile: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
             serverName: kubernetes
 
+## Component scraping kubeDns. Use either this or coreDns
+##
+kubeDns:
```
Comment (tenmozes) [inline charts/victoria-metrics-k8s-stack/values.yaml:742]: could we add docs here?

File: charts/victoria-metrics-k8s-stack/values.yaml [hunk 4]
```diff
@@ -737,6 +737,31 @@ kubeControllerManager:
...
+  service:
+    enabled: false
+    dnsmasq:
+      port: 10054
+      targetPort: 10054
+    skydns:
+      port: 10055
+      targetPort: 10055
+    # selector:
+    #   k8s-app: kube-dns
```
Comment (tenmozes) [inline charts/victoria-metrics-k8s-stack/values.yaml:753]: should selector be uncommenter and be empty?

**Review by tenmozes — CHANGES_REQUESTED** [review]

File: charts/victoria-metrics-k8s-stack/templates/servicemonitors/kubedns.yaml (hunk 1 above)
Comment (f41gh7) [reply charts/victoria-metrics-k8s-stack/templates/servicemonitors/kubedns.yaml]: fixed

File: charts/victoria-metrics-k8s-stack/templates/servicemonitors/kubedns.yaml (hunk 2 above)
Comment (f41gh7) [reply charts/victoria-metrics-k8s-stack/templates/servicemonitors/kubedns.yaml]: replaced with `with`

File: charts/victoria-metrics-k8s-stack/values.yaml (hunk 4 above)
Comment (f41gh7) [reply charts/victoria-metrics-k8s-stack/values.yaml]: yep, it also can be non-empty - with default value.

**Review by tenmozes — APPROVED** [review]

File: charts/victoria-metrics-k8s-stack/templates/servicemonitors/nodeexporter.yaml [hunk 5]
```diff
@@ -9,7 +9,6 @@ spec:
   selector:
     matchLabels:
       app: prometheus-node-exporter
-      app.kubernetes.io/instance: {{ default $.Release.Name $.Values.argocdReleaseOverride | quote }}
```
Comment (k1rk) [inline charts/victoria-metrics-k8s-stack/templates/servicemonitors/nodeexporter.yaml:12]: i remember adding this in #197 but i never tested with this label removed. have you tested with argo @f41gh7 ?

File: charts/victoria-metrics-k8s-stack/values.yaml [hunk 6]
```diff
@@ -737,16 +740,39 @@ kubeControllerManager:
...
 coreDns:
   enabled: true
   service:
     enabled: true
     port: 9153
     targetPort: 9153
-    # selector:
-    #   k8s-app: kube-dns
+    selector:
+       k8s-app: kube-dns
```
Comment (k1rk) [inline charts/victoria-metrics-k8s-stack/values.yaml:775]: 👍

File: charts/victoria-metrics-k8s-stack/templates/servicemonitors/nodeexporter.yaml (hunk 5 above)
Comment (f41gh7) [reply charts/victoria-metrics-k8s-stack/templates/servicemonitors/nodeexporter.yaml:12]: yep, removing it solves scraping issue.

**Review by k1rk — APPROVED** [review]

**Pull request merged**